domain/personallearningplan/src/main/kotlin/uk/gov/justice/digital/hmpps/domain/personallearningplan/Exceptions.kt

```diff
@@ -27,3 +22,28 @@ class ActionPlanAlreadyExistsException(message: String) : RuntimeException(messa
  */
 class GoalNotFoundException(val prisonNumber: String, val goalReference: UUID) :
   RuntimeException("Goal with reference [$goalReference] for prisoner [$prisonNumber] not found")
+
+/**
+ * Thrown when a Goal cannot be arhived/unarchived due to the current state.
+ */
+class InvalidGoalStateException(val prisonNumber: String, val goalReference: UUID, val status: String, val action: String) :
+  RuntimeException("Could not $action goal with reference [$goalReference] for prisoner [$prisonNumber]: Goal was in state [$status] that can't be ${action}d") {
+
+  companion object {
+    const val ARCHIVE = "archive"
+    const val UNARCHIVE = "unarchive"
+  }
+}
+
+/**
+ * A general business exception that takes a string and will have a http code of 409 CONFLICT.
+ */
+
+class BusinessException(val text: String) :
+  RuntimeException(text)
+
+/**
+ * A general not found exception that takes a string and will have a http code of 404 NOT_FOUND.
+ */
+class NotFoundException(val text: String) :
```

Comment:
Kinda related to my comment above, but we've tried to keep our exceptions quite specific, even if that means we end up with a few that ultimately mean a similar thing and are mapped to the same http response code.

`NotFoundException` feels a little vague a generic 🤔 Thinking about how you are currently using this new exception, what are your thoughts on it being called something like `PrisonerHasNoGoalsException` with a constructor arg for the `prisonNumber` rather than the reason text.
```
class PrisonerHasNoGoalsException(val prisonNumber: String) :
  RuntimeException("Prisoner $prisonNumber has no goals")